rmf_traffic/test/unit/agv/test_Planner.cpp

```diff
@@ -2342,3 +2342,588 @@ SCENARIO("Test starts using graph with non-colinear waypoints")
+    }
+    {
+      Eigen::Vector3d robot_loc = {0.1 - 0.1 + 1e-8, 0.1, 0};
+      rmf_utils::optional<Planner::StartSet> start_set_opt =
+          planner.compute_plan_starts(robot_loc, initial_time, 0.1, 1.0);
+      CHECK(start_set_opt);
+      Planner::StartSet start_set = start_set_opt.value();
+      CHECK(!start_set.empty());
+      CHECK(start_set.size() == 1);
+      CHECK(start_set[0].waypoint() == 1);
+    }
+    {
+      Eigen::Vector3d robot_loc = {0.1, 0.1 + 0.1 - 1e-8, 0};
+      rmf_utils::optional<Planner::StartSet> start_set_opt =
+          planner.compute_plan_starts(robot_loc, initial_time, 0.1, 1.0);
+      CHECK(start_set_opt);
+      Planner::StartSet start_set = start_set_opt.value();
+      CHECK(!start_set.empty());
+      CHECK(start_set.size() == 1);
+      CHECK(start_set[0].waypoint() == 1);
+    }
+    {
+      Eigen::Vector3d robot_loc = {0.1, 0.1 - 0.1 + 1e-8, 0};
+      rmf_utils::optional<Planner::StartSet> start_set_opt =
+          planner.compute_plan_starts(robot_loc, initial_time, 0.1, 1.0);
+      CHECK(start_set_opt);
+      Planner::StartSet start_set = start_set_opt.value();
+      CHECK(!start_set.empty());
+      CHECK(start_set.size() == 1);
+      CHECK(start_set[0].waypoint() == 1);
+    }
+  }
+
+  WHEN("Location is near multiple waypoints, could be merged to either one")
+  {
+    graph.add_waypoint(test_map_name, {0, 0}); // 0
+    graph.add_waypoint(test_map_name, {0.01, 0.01}); // 1
+    graph.add_waypoint(test_map_name, {-0.01, -0.01}); // 2
+    graph.add_waypoint(test_map_name, {0.01, -0.01}); // 3
+    graph.add_waypoint(test_map_name, {-0.01, 0.01}); // 4
+    graph.add_waypoint(test_map_name, {10, 10}); // 5
+    REQUIRE(graph.num_waypoints() == 6);
+
+    graph.add_lane(0, 1); // 0
+    graph.add_lane(0, 2); // 1
+    graph.add_lane(0, 3); // 1
+    graph.add_lane(0, 4); // 1
+    graph.add_lane(0, 5); // 1
+    REQUIRE(graph.num_lanes() == 5);
+
+    planner = Planner{Planner::Configuration{graph, traits}, default_options};
+
+    Eigen::Vector3d robot_loc = {0.05, 0.05, 0};
+
+    rmf_utils::optional<Planner::StartSet> start_set_opt =
+        planner.compute_plan_starts(robot_loc, initial_time, 0.1, 1.0);
+    CHECK(start_set_opt);
+
+    Planner::StartSet start_set = start_set_opt.value();
+    CHECK(!start_set.empty());
+    CHECK(start_set.size() == 1);
+    CHECK((start_set[0].waypoint() == 0 || start_set[0].waypoint() == 1
+        || start_set[0].waypoint() == 2 || start_set[0].waypoint() == 3
+        || start_set[0].waypoint() == 4));
+  }
+
+  WHEN("Location is not within merging with waypoints, within lane merging, "
+      "not between lane entry and lane exit, right outside of waypoint "
+      "merging distance.")
+  {
+    graph.add_waypoint(test_map_name, {0, 0}); // 0
+    graph.add_waypoint(test_map_name, {1, 0}); // 1
+    graph.add_waypoint(test_map_name, {10.0, 0}); // 2
+    REQUIRE(graph.num_waypoints() == 3);
+
+    graph.add_lane(0, 1); // 0
```

Comment:
What if we add a lane from 1 -> 0 (`graph.add_lane(1, 0);`)? Every lane that is added is unidirectional, so it would be good to test what happens when there are bidirectional connections between waypoints.

Reply:
gotcha, should have checked what the other tests were doing too :boom: